mypy/plugin.py

```diff
@@ -47,6 +50,21 @@ def analyze_callable_args(self, arglist: TypeList) -> Optional[Tuple[List[Type],
         ('api', TypeAnalyzerPluginInterface)])
 
 
+@trait
+class CommonPluginApi:
```

Comment:
Is this class needed any more?

Reply:
I like to have it for documentation purpose, to separate thing that should be _overridden_ (like `get_xxx_hook()` methods) from things that can be used (like `options` and `lookup_fully_qualified()`). We can however remove it if you don't like it.